appservice/src/createAppService/AppInsightsCreateStep.ts

```diff
@@ -0,0 +1,61 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ApplicationInsightsManagementClient } from 'azure-arm-appinsights';
+import { Location } from 'azure-arm-resource/lib/subscription/models';
+import { AzureWizardExecuteStep, createAzureClient } from 'vscode-azureextensionui';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { nonNullProp, nonNullValue } from '../utils/nonNull';
+import { requestUtils } from '../utils/requestUtils';
+import { IAppServiceWizardContext } from './IAppServiceWizardContext';
+
+export class AppInsightsCreateStep extends AzureWizardExecuteStep<IAppServiceWizardContext> {
+    public priority: number = 500;
+
+    public async execute(wizardContext: IAppServiceWizardContext): Promise<void> {
+        const location: Location = nonNullProp(wizardContext, 'location');
+        const verifyingAppInsightsAvailable: string = localize('verifyingAppInsightsAvailable', 'Verifying that application insights is available for this location...');
+        ext.outputChannel.appendLine(verifyingAppInsightsAvailable);
+        if (await this.appInsightsSupportedInLocation(wizardContext, location)) {
+
+            const creatingNewAppInsights: string = localize('creatingNewAppInsightsInsights', 'Creating new application insights component "{0}"...', wizardContext.newSiteName);
+            ext.outputChannel.appendLine(creatingNewAppInsights);
+
+            const client: ApplicationInsightsManagementClient = createAzureClient(wizardContext, ApplicationInsightsManagementClient);
+            wizardContext.applicationInsights = await client.components.createOrUpdate(
+                nonNullValue(wizardContext.newResourceGroupName),
+                nonNullValue(wizardContext.newSiteName),
```

Comment:
There should be a property for app insights name and you should set it here: https://github.com/microsoft/vscode-azurefunctions/blob/master/src/tree/SubscriptionTreeItem.ts#L138